rules/no-reduce.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+const methodSelector = require('./utils/method-selector');
+const getDocumentationUrl = require('./utils/get-documentation-url');
+
+const message = 'Array.reduce not allowed';
+
+const PROTOTYPE_SELECTOR = [
+	methodSelector({name: 'call'}),
+	'[callee.object.type="MemberExpression"]',
+	'[callee.object.property.type="Identifier"]',
+	'[callee.object.property.name="reduce"]'
```

Comment:
Exclude `computed`(`[][reduce]()`) and limit `callee.object.object` to `Array.prototype` and `[]`.